### PR TITLE
Jcasc configmaps

### PR DESCRIPTION
**Pull request by tnishida1**
* Pulls out jcasc config into their own configmaps so we can manage them
* Creates a configmap to pass in the list of pipelines from the product
* Updates to local terraform so it somewhat works

Edit:
* Updated the pipeline configmap so now jcasc creates pipelines from the product list using the jobdsl plugin

### CODE REVIEW

File: modules/lead/product-jenkins/pipelines.tpl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+%{ for name,pipeline in pipelines ~}
+  ${name}: {
+    type: ${pipeline.type},
+    repo: ${pipeline.repo},
+    org: ${pipeline.org}
+  },
```
Comment (cplee) [inline modules/lead/product-jenkins/pipelines.tpl]: i worry you will end up with invalid JSON due to the extra `,` here

Comment (tnishida1) [reply modules/lead/product-jenkins/pipelines.tpl]: for now I could do something like a substr to remove the extra `,`

Comment (cplee) [reply modules/lead/product-jenkins/pipelines.tpl]: did you test it?  if it works, then i'm good going forward with it as is

Comment (tnishida1) [reply modules/lead/product-jenkins/pipelines.tpl]: yup,  confirmed it works locally and just pushed up the change

File: stacks/environment-aws/main.tf [hunk 2]
```diff
@@ -24,7 +24,7 @@ provider "kubernetes" {
 }
 
 provider "helm" {
-  version         = "1.1.0"
+  version         = "1.0.0"
```
Comment (ChrisSchreiber) [inline stacks/environment-aws/main.tf]: did you mean to roll back this version?

Comment (tnishida1) [reply stacks/environment-aws/main.tf]: was testing some changes, moved the version back to 1.1.0

Comment (mrparkers) [reply stacks/environment-aws/main.tf]: I can approve this once that change is made, everything else LGTM

File: stacks/product-local/variables.tf [hunk 3]
```diff
@@ -40,12 +29,27 @@ variable "enable_keycloak" {
   default = false
 }
 variable "builder_images_version" {
+  default = "v2.0.0"
```
Comment (ChrisSchreiber) [inline stacks/product-local/variables.tf]: not a major issue but I think setting default versions in our stacks creates problems more often than it really helps anything

Comment (tnishida1) [reply stacks/product-local/variables.tf]: gotcha, those were mostly there for testing so I removed this and the jenkins image version default

**Pull request merged**